cartographer/cloud/internal/map_builder_context_interface.h

```diff
@@ -68,6 +75,9 @@ class MapBuilderContextInterface : public async_grpc::ExecutionContext {
       int trajectory_id, LocalSlamSubscriptionCallback callback) = 0;
   virtual void UnsubscribeLocalSlamResults(
       const SubscriptionId& subscription_id) = 0;
+  virtual int SubscribeGlobalSlamOptimizations(
+      GlobalSlamOptimizationCallback callback) = 0;
+  virtual void UnsubscribeGlobalSlamOptimizations(int subscription_index) = 0;
```

Comment:
replace int by SubscriptionId, if you cannot reuse that, rename the latter to LocalSlamSubscriptionId

Reply:
SubscriptionId is not suitable here as it contains a trajectory_id. I will rename to LocalSlamSubscriptionId.